pkg/ccl/streamingccl/streamingest/stream_ingestion_dist.go

```diff
@@ -208,14 +212,162 @@ func startDistIngestion(
 		return rw.Err()
 	}
 
-	updateRunningStatus(ctx, ingestionJob, jobspb.Replicating, "physical replication running")
+	// We now attempt to create initial splits. We currently do
+	// this once during initial planning to avoid re-splitting on
+	// resume since it isn't clear to us at the moment whether
+	// re-splitting is always going to be useful.
+	if !streamProgress.InitialSplitComplete {
+		codec := execCtx.ExtendedEvalContext().Codec
+		splitter := &dbSplitAndScatter{db: execCtx.ExecCfg().DB}
+		if err := createInitialSplits(ctx, codec, splitter, planner.initialTopology, details.DestinationTenantID); err != nil {
+			return err
+		}
+	} else {
+		log.Infof(ctx, "initial splits already complete")
+	}
+
+	if err := ingestionJob.NoTxn().Update(ctx, func(txn isql.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater) error {
+		md.Progress.GetStreamIngest().ReplicationStatus = jobspb.Replicating
+		md.Progress.GetStreamIngest().InitialSplitComplete = true
+		md.Progress.RunningStatus = "physical replication running"
```

Comment:
could splits take a while? Does it make sense to update progress to indicate that "replication is running", then update it again once split concluded?

Reply:
Seems reasonable